sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -164,9 +176,12 @@ private[window] final class SlidingWindowFunctionFrame(
   private[this] var inputLowIndex = 0
 
   /** Prepare the frame for calculating a new partition. Reset all variables. */
-  override def prepare(rows: RowBuffer): Unit = {
+  override def prepare(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
     input = rows
-    nextRow = rows.next()
+    inputIterator = input.generateIterator()
+    if (inputIterator.hasNext) {
```

Comment:
Technically I was keeping it in-par with the earlier code. If there are any `rows` in the buffer, let `nextRow` point to it ... otherwise `nextRow` was already set to `null`. Anyways, I have changed this.